core/storage/btree.rs

```diff
@@ -336,6 +358,65 @@ impl BTreeCursor {
         }
     }
 
+    /// Reads the record of a cell that has overflow pages. This is a state machine that requires to be called until completion so everything
+    /// that calls this function should be reentrant.
+    fn process_overflow_read(
+        &mut self,
+        payload: &'static [u8],
+        start_next_page: u32,
+        payload_size: u64,
+    ) -> Result<Option<Record>> {
```

Comment:
IMO, like in other places in Limbo code, this should return some sort of enum with `::IO` in the "not ready" cases, I don't really like `Option` for these semantics because it obfuscates whether it means "not ready" or "there was no data" -- even if in this _particular_ case it always returns data when it's ready. I don't like using Option for this kind of thing in Limbo and I don't like it in you-know-what either 😂